tutorials/tutorial-howto-simulate-iot-sensor-data.md

```diff
@@ -0,0 +1,269 @@
+> - [Prerequisites](#prereqs)
+> - [Step 1 - Set up your tables](#step1)
+> - [Step 2 - Create the simulated IoT sensor data](#step2)
+> - [Step 3 - Run basic queries (optional)](#step3)
+> - [Next Steps](#next-steps)
+
+### Introduction [](introduction)
+The Internet of Things (IoT) describes a trend where computing is 
+becoming ubiquitous and is embedded in more and more physical things. 
+For many of these things, the purpose of IoT is to collect sensor data 
+about the environment in which it exists: e.g., oil wells, factories, 
+power plants, farms, moving vehicles, office buildings, homes.
+
+In other words, IoT is all about the data. And the datasets generated by 
+these things are generally time-series in nature, with relational 
+metadata to describe those things.
+
+Often, it is necessary to simulate IoT sensor data, for example, when
+testing a new system. In this tutorial, we show how to simulate a 
+basic IoT sensor dataset on PostgreSQL, or TimescaleDB.
+
+*Note: For creating a more advanced simulated IoT dataset, please
+try the Time-series Benchmarking Suite (TSBS) ([Github][github-tsbs]).*
+
+### Prerequisites [](prereqs)
+
+* A PostgreSQL or TimescaleDB
+([installation instructions][get-timescale-cloud]) instance
+
+* `psql` ([installation instruction][get-psql])
+
+>:TIP: If you install PostgreSQL or TimescaleDB locally, `psql` may come
+> preinstalled
+
+### Step 1 - Set up your tables [](step1)
+First, connect to your database via `psql`.
+
+Second, create the “sensors” and “sensor_data” tables:
+
+```sql
+CREATE TABLE sensors(
+  id SERIAL PRIMARY KEY,
+  type VARCHAR(50),
+  location VARCHAR(50)
+);
+```
+
+```sql
+CREATE TABLE sensor_data (
+  time TIMESTAMPTZ NOT NULL,
+  sensor_id INTEGER,
+  temperature DOUBLE PRECISION,
+  cpu DOUBLE PRECISION,
+  FOREIGN KEY (sensor_id) REFERENCES sensors (id)
+);
+```
+
+Third (only if you are using TimescaleDB), convert the `sensor_data`
+table into a [`hypertable`][docs-hypertable]:
+
+```sql
+SELECT create_hypertable('sensor_data', 'time');
+```
+
+Fourth, populate the sensors table with 4 sensors:
+
+```sql
+INSERT INTO sensors (type, location) VALUES
+('a','floor'),
+('a', 'ceiling'),
+('b','floor'),
+('b', 'ceiling');
+```
+
+Fifth, verify that the sensors were created correctly:
+
+```sql
+SELECT * FROM sensors;
+```
+
+After running that last SQL statement, you should see something like
+this:
+
+```
+ id | type | location 
+----+------+----------
+  1 | a    | floor
+  2 | a    | ceiling
+  3 | b    | floor
+  4 | b    | ceiling
+(4 rows)
+```
+
+### Step 2 - Create the simulated IoT sensor data [](step2)
+
+*Note: for the following sections we’ll share the results of our queries as an example, but since the tutorials generates random data every time it is run, your results will look different (but will be structured the same way).*
+
+First, generate a dataset of simulated data for one sensor, recording data every 5 minutes for the past 24 hours:
```

Comment:
I think this query can probably be removed now or needs some adjustment after the change to the other generator queries.

Reply:
Updated